configuration/src/main/java/io/airlift/configuration/ConditionalModule.java

```diff
@@ -49,26 +50,51 @@ public static <T> Module conditionalModule(Class<T> config, Predicate<T> predica
                 conditionalModule(config, predicate.negate(), otherwise));
     }
 
+    public static <T> Module conditionalModule(Class<T> config, String prefix, Predicate<T> predicate, Module module, Module otherwise)
+    {
+        return combine(
+                conditionalModule(config, prefix, predicate, module),
+                conditionalModule(config, prefix, predicate.negate(), otherwise));
+    }
+
     public static <T> Module conditionalModule(Class<T> config, Predicate<T> predicate, Module module)
     {
-        return new ConditionalModule<>(config, predicate, module);
+        return conditionalModule(config, Optional.empty(), predicate, module);
+    }
+
+    public static <T> Module conditionalModule(Class<T> config, String prefix, Predicate<T> predicate, Module module)
+    {
+        return conditionalModule(config, Optional.of(prefix), predicate, module);
+    }
+
+    public static <T> Module conditionalModule(Class<T> config, Optional<String> prefix, Predicate<T> predicate, Module module)
```

Comment:
should this be private?

Reply:
Yeah, using a prefix is never dynamically optional, so this doesn't need to be in the API. We should simply inline this method into the callers, since it is trivial.